packages/l1-l3-teleport/scripts/initiate-deposit.ts

```diff
@@ -0,0 +1,151 @@
+import { providers, Wallet, BigNumber, ethers } from "ethers"
+import { arbLog, requireEnvVariables } from 'arb-shared-dependencies'
+import { Erc20L1L3Bridger, getL2Network } from "godzillaba-arbitrum-sdk"
+import { ERC20__factory } from "godzillaba-arbitrum-sdk/dist/lib/abi/factories/ERC20__factory"
+import yargs from 'yargs/yargs'
+import { hideBin } from "yargs/helpers"
+
+// Importing configuration //
+require('dotenv').config()
+requireEnvVariables(['L1RPC', 'L2RPC', 'L3RPC', 'L1PRIVKEY'])
+
+// Initial setup //
+const l1Provider = new providers.JsonRpcProvider(process.env.L1RPC)
+const l2Provider = new providers.JsonRpcProvider(process.env.L2RPC)
+const l3Provider = new providers.JsonRpcProvider(process.env.L3RPC)
+
+const l1Signer = new Wallet(process.env.L1PRIVKEY!, l1Provider)
+
+const main = async (params: {
+  l1Token: string, amount: BigNumber, l3Recipient?: string, skipFeeToken?: boolean
+}) => {
+  await arbLog(`Bridging tokens from L1 to L3`)
+
+  /**
+   * Use L3 Network to initialize a bridger
+   */
+  const l3Network = await getL2Network(l3Provider)
+  const bridger = new Erc20L1L3Bridger(l3Network)
+
+  /**
+   * Information about the configuration of the bridger
+   * 
+   * Note that if the L3's fee token is not ETH you must pay for the L2->L3 fee in the L3's fee token.
+   * In this case you may either:
+   * A) Pay the fee in the L3's fee token when initiating the deposit on L1. 
+   *    This can only be done if the fee token is available and held on L1.
+   *    This is the default behavior, and no additional steps are required besides approving the fee token.
+   * B) Manually redeem the retryable on L3 to pay the fee (this can be done by anyone, not just the sender of the deposit). 
+   *    If the fee token is unavailable on L1 or skipFeeToken is set to true, manual redemption is required.
+   * 
+   * If the L3 uses ETH for fees, all retryables are paid for when initiating the deposit by default and no additional steps are required.
+   */
+  const l1ChainId = (await l1Provider.getNetwork()).chainId
+  const l2ChainId = (await l2Provider.getNetwork()).chainId
+  const l3ChainId = (await l3Provider.getNetwork()).chainId
+  const signerAddress = await l1Signer.getAddress()
+  const l3FeeTokenL1Address = await bridger.l1FeeTokenAddress(l1Provider, l2Provider)
+  const feeTokenSymbol = bridger.l2FeeTokenAddress ? await ERC20__factory.connect(bridger.l2FeeTokenAddress, l2Provider).symbol() : 'ETH'
+  console.log('L1 Token:', params.l1Token)
+  console.log('L3 Fee Token:', feeTokenSymbol || 'ETH')
+  console.log('L3 Fee Token Address on L2:', bridger.l2FeeTokenAddress || ethers.constants.AddressZero)
```

Comment:
Should we show 'ETH' here instead of AddressZero?

Reply:
384b9900e2ba0d656b4b9d5e96131471ec608584